Revit_Engine/Query/IncludeNonVisibleObjects.cs

```diff
@@ -35,17 +35,19 @@ public static partial class Query
         /**** Public Methods                            ****/
         /***************************************************/
 
-        [Description("Returns true if FilterRequest should include non visible objects when pulling edges")]
-        [Input("filterRequest", "FilterRequest")]
+        [Description("Returns true if IRequest should include non visible objects when pulling edges")]
+        [Input("request", "IRequest")]
         [Output("IncludeNonVisibleObjects")]
-        public static bool IncludeNonVisibleObjects(this FilterRequest filterRequest)
+        public static bool IncludeNonVisibleObjects(this IRequest request)
         {
-            if (filterRequest == null)
+            if (request == null || !(request is FilterRequest))
                 return false;
 
-            if (filterRequest.Equalities.ContainsKey(Convert.FilterRequest.IncludeNonVisibleObjects))
+            FilterRequest aFilterRequest = (FilterRequest)request;
```

Comment:
I don't see the point of this. Why defining the method for a generic IRequest if then it works only with FilterRequest?
I'd this method needs to be changed to either to accept a new, appropriate implementation of IRequest.

It's particularly important here because the `Equalities` Dictionary property will be soon removed from the FilterRequest.